backend/src/api/transactions/transaction.model.js

```diff
@@ -28,13 +28,49 @@ const transactionSchema = new Schema(
 );
 
 transactionSchema.statics.createTransaction = createTransaction;
+transactionSchema.statics.getCurrentMonthBalance = getCurrentMonthBalance;
 
 const TransactionModel = mongoose.model('Transaction', transactionSchema);
 
 async function createTransaction(info) {
   return this.create(info);
 }
 
+async function getCurrentMonthBalance(userId) {
+  const date = new Date();
+  const startMonth = new Date(date.getFullYear(), date.getMonth(), 1);
+  console.log('startMonth', startMonth);
+  const expenseGroup = await this.aggregate([
+    {
+      $match: {
+        userId,
+        transactionDate: { $gte: startMonth },
+      },
+    },
+    {
+      $addFields: {
+        expense: {
+          $cond: [{ $eq: ['$type', 'EXPENSE'] }, '$amount', 0],
+        },
+        income: {
+          $cond: [{ $eq: ['$type', 'INCOME'] }, '$amount', 0],
+        },
+      },
+    },
+    {
+      $group: {
+        _id: null,
+        expense: { $sum: '$expense' },
+        income: { $sum: '$income' },
+        monthBalance: { $sum: { $subtract: ['$income', '$expense'] } },
```

Comment:
We won't have INCOME transaction for current month till the end of it - that's the rules ) You'll need to get it manually from user document (salary + passive income).

Reply:
Done